pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/KubernetesRuntimeFactory.java

```diff
@@ -185,14 +185,16 @@ public boolean externallyManaged() {
     public KubernetesRuntime createContainer(InstanceConfig instanceConfig, String codePkgUrl,
                                              String originalCodeFileName,
                                              Long expectedHealthCheckInterval) throws Exception {
-        String instanceFile;
+        String instanceFile = null;
         switch (instanceConfig.getFunctionDetails().getRuntime()) {
             case JAVA:
                 instanceFile = javaInstanceJarFile;
                 break;
             case PYTHON:
                 instanceFile = pythonInstanceFile;
                 break;
+            case GO:
```

Comment:
is this supported? if it is not fully supported, I would suggest throwing UnsupportedOperationException in this place until we confirm it is working.